client/src/features/movie-search/container.js

```diff
@@ -1,58 +1,78 @@
 import { connect } from 'react-redux';
 import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
 import { bindActionCreators } from 'redux';
 import PropTypes from 'prop-types';
-import * as Action from './actions';
-
-import MovieHeader from '../header';
-import MovieFooter from '../footer';
-import MovieList from '../movie-list';
-import { getFilms, getFilmsWithParams } from './utils';
-import { getValueFilter } from '../../core/store/selectors';
-
-class SeriesContainer extends Component {
-  async componentDidMount() {
-    const { setMovies } = this.props;
-    const { data } = await getFilms();
-    setMovies(data.data);
+import * as Action from '../common/actions';
+import { getFilmsWithQuery } from '../common/utils';
+import styles from './component.less';
+import Logo from '../../shared/logo';
+import Filter from './search-filter';
+import MovieFooter from '../../shared/footer';
+import MovieList from '../../shared/movie-list';
+import { getVisibleFilmsLength } from '../../core/store/selectors';
+
+class MovieSearchContainer extends Component {
+  static propTypes = {
+    setMovies: PropTypes.func.isRequired,
+    filmsLength: PropTypes.number,
+    location: PropTypes.objectOf(PropTypes.string).isRequired,
+    history: PropTypes.objectOf(PropTypes.any).isRequired,
+  };
+
+  componentDidMount() {
+    this.componentAddItems();
   }
 
-  async componentDidUpdate(prevProps) {
-    const { filter, setMovies } = this.props;
-    if (filter !== prevProps.filter) {
-      if (filter.text !== '') {
-        const { data } = await getFilmsWithParams(filter, filter.text);
-        setMovies(data.data);
-      }
+  componentDidUpdate(prevProps) {
+    const { location } = this.props;
+    if (location.pathname !== prevProps.location.pathname) {
+      this.componentAddItems();
+    }
+  }
+
+  async componentAddItems() {
+    const { history, setMovies } = this.props;
+    let url = history.location.pathname;
+    if (url !== '/search') {
+      url = url.substring(15);
```

Comment:
deal with route parameters - take them implicitly from `this.props.match.params`, pass this object into `getFilmsWithQuery` and build a query string there